### DISCUSSION TITLE
module 'torch' has no attribute 'sparse_csr'

### DISCUSSION THREAD

**Discussion by Jessi5177** [pyg-team/pytorch_geometric#7189] [Q&A]
my code:

```
class GatConv(MessagePassing):
    def __init__(self, in_channels, out_channels, edge_channels,
                 negative_slope=0.2,dropout=0):
        super(GatConv, self).__init__(aggr='add')
                                ...
    def forward(self, x, edge_index, edge_attr, size=None):
        x = self.fc(x)
        return self.propagate(edge_index, size=size, x=x, edge_attr=edge_attr)
```

 Version：

PyTorch version: 1.8.1+cu111
torch-cluster    1.5.9    pypi_0    pypi
torch-geometric    2.3.0     pypi_0    pypi
torch-scatter    2.0.6          pypi_0    pypi
torch-sparse     0.6.9          pypi_0    pypi
torch-spline-conv    1.2.1    pypi_0    pypi

**Comment by akihironitta** [comment]
Looks like `torch.sparse_csr` is available only from PyTorch 1.9.0 and onwards https://github.com/pytorch/pytorch/commit/5fb1142702320f0d52a3d87a94ab4c93220013c9.  Also, maybe one of the maintainers can confirm, but from what I see in the git log, PyG 2.3.0 was never tested with PyTorch 1.8.1.

**Reply by rusty1s** [reply]
Yes, PyG 2.3 at least requires PyTorch 1.10.

**Reply by Jessi5177** [reply]
Does that mean I can only update pytorch to 1.10.0?

**Reply by rusty1s** [reply]
For latest PyG, yes.

**Reply by Jessi5177** [reply]
Could I use the old version of PyG to call GatConv? If so, which version of my current function package needs to be reduced?

**Reply by rusty1s** [reply]
You can try downgrading PyG to 2.1 or 2.2, no other package version needs to be reduced.
